### PR TITLE
Add intraday heartrate by interval api function

### PR DESCRIPTION
**Pull request by andrewKOwong**

### CODE REVIEW

File: core/api.py [hunk 1]
```diff
@@ -119,3 +137,76 @@ def by_date(self, date:str = 'today', period: str = '1d' ) -> dict:
...
+        '2022-08-14' | '2022-08-14' | '1min        | '07:00'    | '13:00'  | daily + timeboxed intraday
+        '2022-08-14' | '2022-08-14' | '1sec'       | '07:00'    | '13:00'  | daily + timeboxed intraday at 1 sec intervals
+        '2022-08-11' | '2022-08-14' | '1min        | '07:00'    | '13:00'  | invalid
+        '2022-08-11' | '2022-08-14' | '1sec'       | '07:00'    | '13:00'  | invalid
+
+        Reference:
+        https://dev.fitbit.com/build/reference/web-api/heartrate-timeseries/get-heartrate-timeseries-by-date/
+
+        Args:
+            start_date (str, optional): yyyy-MM-dd format date, or 'today'.. Defaults to 'today'.
```
Comment (ursulahneumann) [inline core/api.py]: Are the double periods are here and on the next line a mistake?

File: core/api.py [hunk 2]
```diff
@@ -119,3 +137,76 @@ def by_date(self, date:str = 'today', period: str = '1d' ) -> dict:
...
+        """Heart rate intraday by interval.
+
+        Time parameters are optional.
+
+        API endpoint returns inconsistent results depending on the parameters.
+        The following table documents the results of different parameter combinations:
+
+        start_date   | end_date     | detail_level | start_time | end_time | result
+        ---------------------------------------------------------------------
+        'today'      | 'today'      | '1min'       | None       | None     | daily + intraday
```
Comment (ursulahneumann) [inline core/api.py]: For the result column where it says intraday (but not intraday at 1 sec intervals), should it say intraday at 1 min intervals so it's consistent?

File: core/api.py [hunk 3]
```diff
@@ -119,3 +137,76 @@ def by_date(self, date:str = 'today', period: str = '1d' ) -> dict:
...
+        # Validate detail level
+        DETAIL_LEVELS = ['1sec', '1min']
+        if detail_level not in DETAIL_LEVELS:
+            raise ValueError(f"detail_level {detail_level} should be one of {DETAIL_LEVELS}.")
+
+        # Validate date
+        start_date = check_date_format_wrapper(start_date)
+        end_date = check_date_format_wrapper(end_date)
+
+        # Validate both times or neither provided
```
Comment (ursulahneumann) [inline core/api.py]: This comment is a big vague, I misread it the first time.  Can you make it more clear by saying "Validate both start and end times, or whether neither time was provided" or something?

**Review by ursulahneumann — CHANGES_REQUESTED** [review]

File: core/api.py (hunk 1 above)
Comment (andrewKOwong) [reply core/api.py]: done

File: core/api.py (hunk 2 above)
Comment (andrewKOwong) [reply core/api.py]: done

File: core/api.py (hunk 3 above)
Comment (andrewKOwong) [reply core/api.py]: done

**Pull request merged**